.github/workflows/build-wheels.yml

```diff
@@ -0,0 +1,197 @@
+
+jobs:
+  determine-source-date-epoch:
+    name: "Determine SOURCE_DATE_EPOCH"
+    runs-on: ubuntu-latest
+    outputs:
+      source-date-epoch: ${{ steps.log.outputs.source-date-epoch }}
+    if: github.repository_owner == 'scikit-hep'
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - id: log
+        name: Compute SOURCE_DATE_EPOCH
+        run: |
+          # Find latest unix timestamp in awkward-cpp, and the kernel generation files
+          epoch=$( git log -1 --format=%at -- awkward-cpp kernel-specification.yml kernel-test-data.json )
+          echo "source-date-epoch=$epoch" >> $GITHUB_OUTPUT
+
+  make_sdist:
+    name: "Build awkward-cpp sdist"
+    runs-on: ubuntu-latest
+    needs: [determine-source-date-epoch]
+    env:
+      SOURCE_DATE_EPOCH: ${{ needs.determine-source-date-epoch.outputs.source-date-epoch }}
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Prepare build files
+      run: pipx run nox -s prepare
+
+    - name: Build awkward-cpp sdist
+      run: pipx run build --sdist awkward-cpp
+
+    - name: Check metadata
+      run: pipx run twine check awkward-cpp/dist/*
+
+    - uses: actions/upload-artifact@v4
+      with:
+        name: awkward-cpp-sdist
+        path: awkward-cpp/dist/*.tar.gz
+
+  build_wheels:
+    needs: [determine-source-date-epoch]
+    name: "Wheel awkward-cpp: ${{ matrix.type }} ${{ matrix.arch }} on ${{ matrix.os }} with ${{ matrix.build }}"
+    runs-on: ${{ matrix.os }}
+    env:
+      SOURCE_DATE_EPOCH: ${{ needs.determine-source-date-epoch.outputs.source-date-epoch }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        arch: [auto64]
+        build: ["cp", "pp"]
+
+        include:
+        - os: macos-latest
+          type: "Universal"
+          arch: universal2
+          build: "cp"
+
+        - os: windows-latest
+          arch: auto64
+          build: "cp"
+
+        - os: windows-latest
+          arch: auto32
+          build: "cp{38,39}-"
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Python 3.11
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+
+    - name: Prepare build files
+      run: pipx run nox -s prepare
+
+    - uses: pypa/cibuildwheel@v2.16.5
+      env:
+        CIBW_BUILD: "${{ matrix.build }}*"
+        CIBW_ARCHS: ${{ matrix.arch }}
+      with:
+        config-file: cibuildwheel.toml
+        package-dir: awkward-cpp
+
+    - name: Check metadata
+      shell: python
+      run: |
+        import subprocess, glob
+        subprocess.run(
+          ["pipx", "run", "twine", "check", *glob.glob("wheelhouse/*.whl")],
+          check=True
+        )
+
+    - name: Upload wheels
+      uses: actions/upload-artifact@v4
+      with:
+        name: awkward-cpp-wheels-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.build }}
```

Comment:
Using `awkward-cpp-wheels` here (and for the rest of the workflow) to match search pattern in 

https://github.com/scikit-hep/awkward/blob/b7eb8b9a0d2fb801b2aebb89620cd3fdab1339d6/.github/workflows/upload_nightly_wheels.yml#L34